### ISSUE TITLE
Requests: various 

### ISSUE THREAD

**Issue by brendan-pike** [Marekkon5/discogstagger#5]
Firstly this is amazing, I've been looking so long (linux) for something that can properly handle discog genre/style.  I'm just running it through my collection (99872 tracks) now.

- Better explanation on some of the options, eg. will checking overwrite tags remove other tags, will all cover art be removed if there are multiple images, 
- I'd love to also populate any other tags like engineers, instruments, label, mixed at, pretty much what ever is available. eg. performer[drums]john smith etc. I generally use picard / musicbrainz manually for this.
- I prefer to delete any embedded art and put them it in the album directory, cover.jpg, back,jpg etc. so an option to do this would be great
- I occasionally need to use custom/non-typical tags in genre/style. It would be great if there was a preserve list so they wouldn't be removed.
- Lastly more of a query, does the master release generally provide more accurate tagging of genre/style, I notice the versions can get pretty messy at times, therefore perhaps allow option to give preference to the master release genre/style?

**Comment by Marekkon5** [comment]
1. I tought it was pretty self-explanatory, but: overwrite will only overwrite the tags you've selected. If overwrite is not enabled, the selected tags will be only written if the target tag is empty. As for album art it works the same, but only for front covers - rest of the images should be intact.
2. As for the different artist types (performer/producer/design...) they are currently included within all the artists. Can add an option to separate to own tags in next update, however not all of them are available for every track. The other tags (mixed at, instruments) I don't think are available on Discogs.
3. Can add next update to not embed art, but save as file.
4. Do you mean option to append the Discogs genres/styles to the current values that are in the tag?
5. Right now the tagger first tries to match masters, then falls back to releases, so I guess it already should use genres/styles from masters.

**Comment by brendan-pike** [comment]
Clarifying 4.  eg. existing genre might be "Rock;  MyCustom; Folk", so if I list the word MyCustom in the preserve list then I'm happy for Rock and Folk to be removed / changed but would like MuCustom to remain in the tag.

**Comment by Marekkon5** [comment]
Yes, but what should happen with the genres that were gathered from Discogs?

**Comment by brendan-pike** [comment]
They should be inserted alongside my Custom one, so in my above example lets say Folk was not found in discog but Rock and Fusion was then I would expect the result to look like this, "Rock; MyCustom; Fusion" with the Folk tag dropped since it was neither in my preserve list nor found at discog.

**Comment by brendan-pike** [comment]
Just adding to point 2. I see that discog does have information like instrument players, writer, producers, credits, engineers etc, Here is an example https://www.discogs.com/Waldeck-Atlantic-Ballroom/release/12712540 but I don't know if this information is available via the API

**Comment by djmufasa** [comment]
This is a neat tool. Thanks to the developer for sharing. 

2 Requests - 

1. Is it possible to have a genre append feature - eg add discogs genre/style to content in the track genre tags instead of overwriting.
2. 
3. Is it possible to add the write/append the "Style" to the "comments" tag as well

**Comment by bascurtiz** [comment]
@djmufasa 
1. @Marekkon5 will take care of that append feature soon.
2. If you use mp3tag (both available for macOS + Windows) you can append Style to Comment yourself.
If you use windows, I've attached the Action for it (perhaps works on macOS aswell, havent checked)
a. unzip into %APPDATA%\Mp3tag\data\actions
b. open mp3tag, add the tracks you want to apply the action to and select them. 
c. hit Actions on top and click Append Style to Comment.
[Append Style to Comment.zip](https://github.com/Marekkon5/discogstagger/files/6208642/Append.Style.to.Comment.zip)
